gemrb/core/Map.cpp

```diff
@@ -1704,8 +1712,16 @@ void Map::InitActor(Actor *actor)
 	//guess game is always loaded? if not, then we'll crash
 	ieDword gametime = core->GetGame()->GameTime;
 
-	if (IsVisible(actor->Pos, false) && actor->Schedule(gametime, true) ) {
-		ActorSpottedByPlayer(actor);
+	if (IsVisible(actor->Pos, false)) {
+		// This case is probably not possible.  At this point, the actor's position
+		// ought to be 0,0 and the caller of AddActor is expected to follow up with
+		// SetPosition to move the actor to its real position.
+		// The Schedule call here preserves previous behaviour, while also logging
+		// the event if something unexpected happens.  It's unclear whether an
+		// actor can be visible at 0,0 by accident and whether the Schedule call
+		// is meaningful for that case.
+		Log(ERROR, "Map", "Actor visible in InitActor before real position was set");
+		actor->Schedule(gametime, true);
```

Comment:
This is a noop, since you're not using the return value (Schedule is const). I question the need for the whole block.

Reply:
Ah, ok. The name Schedule suggested that an action takes place. The whole block can really be deleted then.